### PR TITLE
Use primary color property throughout the app 

### PR DESCRIPTION
**Pull request by andreeaceachir142**
# What 
- Use primary color property throughout the app.

# How 
- Use Recoil to store the `primaryColor` property in an atom and use it throughout the app to update the UI elements which should have the color changed.
- Components that should be updated: the spinner on the splash screen, the button on the location details page, the button on the wayfinding page, the accessibility toggle button and the chip components.

### CODE REVIEW

File: packages/components/src/components/chip/chip.tsx [hunk 1]
```diff
@@ -8,35 +8,45 @@ import { JSX } from '@stencil/core/internal';
...
+     */
+
+    @Prop() content: string;
+
+    /**
+     * The primary color of the chip.
+     *
+     * @type {string}
+     */
+    @Prop() primaryColor?: string;
```
Comment (ammapspeople) [inline packages/components/src/components/chip/chip.tsx:31]: It's used for background color only. Should we call it that instead?
And should we write what the default value is?

File: packages/map-template/src/App.jsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ const mapboxAccessToken = params.get('mapboxAccessToken');
 
 // Append the hashtag symbol to the color code (i.e. ffffff)
 const primaryColor = params.get('primaryColor');
-const hexPrimaryColor = primaryColor ? '#'.concat(primaryColor) : undefined;
+const hexPrimaryColor = primaryColor ? '#'.concat(primaryColor) : '#005655';
```
Comment (ammapspeople) [inline packages/map-template/src/App.jsx:21]: We still don't have a way to use MIDT in JavaScript?
If not, write a comment about which MIDT color this is.

File: packages/map-template/src/atoms/primaryColorState.js [hunk 3]
```diff
@@ -0,0 +1,9 @@
+import { atom } from 'recoil';
+
+// The HEX value refers to the --brand-colors-dark-pine-100 from MIDT
+const primaryColorState = atom({
+    key: 'primaryColor',
+    default: '#005655'
```
Comment (ammapspeople) [inline packages/map-template/src/atoms/primaryColorState.js:6]: We still don't have a way to use MIDT in JavaScript?
If not, write a comment about which MIDT color this is.

File: packages/map-template/src/components/BottomSheet/BottomSheet.jsx [hunk 4]
```diff
@@ -20,6 +20,7 @@ import LocationsList from '../LocationsList/LocationsList';
  * @param {function} props.pushAppView - Function to push to app view to browser history.
  * @param {string} props.currentAppView - Holds the current view/state of the Map Template.
  * @param {array} props.appViews - Array of all possible views.
+ *
```
Comment (ammapspeople) [inline packages/map-template/src/components/BottomSheet/BottomSheet.jsx:23]: Is this your editor again? 😛

File: packages/map-template/src/components/LocationDetails/LocationDetails.jsx [hunk 5]
```diff
@@ -195,7 +199,7 @@ function LocationDetails({ onBack, onStartWayfinding, onSetSize, snapPointSwiped
                 </section>}
             </div>
 
-            <button onClick={() => startWayfinding()} className="location-details__wayfinding">
+            <button onClick={() => startWayfinding()} style={{background: primaryColor}} className="location-details__wayfinding">
```
Comment (ammapspeople) [inline packages/map-template/src/components/LocationDetails/LocationDetails.jsx:202]:
```suggestion
            <button onClick={() => startWayfinding()} style={{ background: primaryColor }} className="location-details__wayfinding">
```

File: packages/map-template/src/components/MapTemplate/MapTemplate.jsx [hunk 6]
```diff
@@ -182,10 +185,20 @@ function MapTemplate({ apiKey, gmApiKey, mapboxAccessToken, venue, locationId, p
...
     useEffect(() => {
         setTileStyle(tileStyle)
     }, [tileStyle]);
 
+     /*
+     * React on changes in the primary color prop.
+     */
+     useEffect(() => {
+        setPrimaryColor(primaryColor)
+    }, [primaryColor]);
```
Comment (ammapspeople) [inline packages/map-template/src/components/MapTemplate/MapTemplate.jsx:200]:
```suggestion
    /*
     * React on changes in the primary color prop.
     */
    useEffect(() => {
        setPrimaryColor(primaryColor)
    }, [primaryColor]);
```

File: packages/map-template/src/components/BottomSheet/BottomSheet.jsx (hunk 4 above)
Comment (andreeaceachir142) [reply packages/map-template/src/components/BottomSheet/BottomSheet.jsx:23]: Noooo.. 🫣

Comment (andreeaceachir142) [reply packages/map-template/src/components/BottomSheet/BottomSheet.jsx:23]: Fixed it 😁

File: packages/map-template/src/atoms/primaryColorState.js (hunk 3 above)
Comment (andreeaceachir142) [reply packages/map-template/src/atoms/primaryColorState.js:6]: No, unfortunately we don't have a way to use MIDT in JS. There is already a comment about which color it refers to. 😁

File: packages/map-template/src/App.jsx (hunk 2 above)
Comment (andreeaceachir142) [reply packages/map-template/src/App.jsx:21]: Added comment about HEX value.

File: packages/components/src/components/chip/chip.tsx (hunk 1 above)
Comment (andreeaceachir142) [reply packages/components/src/components/chip/chip.tsx:31]: Yeah, good idea. Renamed it to `backgroundColor` instead, and added default color as well

File: packages/map-template/src/atoms/primaryColorState.js (hunk 3 above)
Comment (ammapspeople) [reply packages/map-template/src/atoms/primaryColorState.js:6]: Lol, didn't see it 😄

File: packages/components/src/components/chip/chip.tsx [hunk 7]
```diff
@@ -8,35 +8,46 @@ import { JSX } from '@stencil/core/internal';
...
+     * The chip content that is displayed in the component.
+     *
+     * @type {string}
+     */
+
+    @Prop() content: string;
+
+    /**
+     * The background color of the chip.
+     * The default HEX value refers to the --brand-colors-dark-pine-100 from MIDT
```
Comment (ammapspeople) [inline packages/components/src/components/chip/chip.tsx:28]: Write the hex value as well, so customers don't have to look up in MIDT, which they may not know about.

Comment (andreeaceachir142) [reply packages/components/src/components/chip/chip.tsx]: ✅

**Review by ammapspeople — APPROVED** [review]
👏

**Pull request merged**